modules/programs/direnv.nix

```diff
@@ -104,11 +104,13 @@ in {
       source = tomlFormat.generate "direnv-config" cfg.config;
     };
 
-    xdg.configFile."direnv/direnvrc" = let
-      text = concatStringsSep "\n" (optional (cfg.stdlib != "") cfg.stdlib
-        ++ optional cfg.nix-direnv.enable
-        "source ${cfg.nix-direnv.package}/share/nix-direnv/direnvrc");
-    in mkIf (text != "") { inherit text; };
+    xdg.configFile."direnv/lib/hm-nix-direnv.sh" = mkIf cfg.nix-direnv.enable {
+      source = "${cfg.nix-direnv.package}/share/nix-direnv/direnvrc";
+      executable = true;
```

Comment:
Nit: I don't think this is necessary.

Reply:
I thought lib files had to end in `.sh` _and_ be executable, but only the first condition is actually needed. I'll remove it and squash everything once I get back on my PC